data.sql

```diff
@@ -17,4 +17,167 @@ insert into animals(name,date_of_birth, weight_kg,neutered,escape_attempts) valu
+('Bob', 45),
+('Melody Pond', 77),
+('Dean Winchester', 14),
+('Jodie Whittaker', 38);
+select * from owners;
+commit;
+select * from owners;
+
+begin;
+insert into species (name) values
+('Pokemon'),
+('Digimon');
+select * from species;
+commit;
+select * from species;
+
+begin;
+update animals set species_id=2 where name like '%mon';
+select * from animals;
+commit;
+select * from animals;
+
+begin;
+update animals set species_id=1 where species_id is null;
+select * from animals;
+commit;
+select * from animals;
+
+begin;
+select * from owners;
+update animals set owner_id=1 where name='Agumon' returning *;
+update animals set owner_id=2 where name='Gabumon' and name='Pikachu' returning *;
+update animals set owner_id=2 where name='Gabumon' or name='Pikachu' returning *;
+update animals set owner_id=3 where name='Plantmon' or name='Devimon' returning *;
+update animals set owner_id=4 where name='Charmander' or name='Squirtle' or name='Blossom' returning *;
+update animals set owner_id=5 where name='Angemon' or name='Boarmon' returning *;
+select * from animals;
+commit;
+select * from animals;
+
+-- Day 4 
+
+begin;
+
+insert into vets
+(name, age, date_of_graduation) values
+('William Tatcher',45,'2000-04-23'),
+('Maisy Smith',26,'2019-01-17'),
+('Stephanie Mendez', 64, '1981-05-04'),
+('Jack Harkness',38,'2008-06-08');
+
+select * from vets;
+
+--  id |       name       | age | date_of_graduation 
+-- ----+------------------+-----+--------------------
+--   5 | William Tatcher  |  45 | 2000-04-23
+--   6 | Maisy Smith      |  26 | 2019-01-17
+--   7 | Stephanie Mendez |  64 | 1981-05-04
+--   8 | Jack Harkness    |  38 | 2008-06-08
+-- (4 rows)
+
+commit;
+
+select * from vets;
+
+--  id |       name       | age | date_of_graduation 
+-- ----+------------------+-----+--------------------
+--   5 | William Tatcher  |  45 | 2000-04-23
+--   6 | Maisy Smith      |  26 | 2019-01-17
+--   7 | Stephanie Mendez |  64 | 1981-05-04
+--   8 | Jack Harkness    |  38 | 2008-06-08
+-- (4 rows)
+
+begin;
+
+insert into specializations
+(vet_id,specie_id) values
+(5,1),
+(7,1),
+(7,2),
+(8,2);
+
+select * from specializations;
+--  vet_id | specie_id 
+-- --------+-----------
+--       5 |         1
+--       7 |         1
+--       7 |         2
+--       8 |         2
+-- (4 rows)
+
+commit;
+
+select * from specializations;
+--  vet_id | specie_id 
+-- --------+-----------
+--       5 |         1
+--       7 |         1
+--       7 |         2
+--       8 |         2
+-- (4 rows)
+
+
+begin;
+
+
+insert into visits (animal_id,vet_id,date_of_visit) values
+(1,5,'2020-5-24'),
+(1,7,'2020-07-22'),
+(2,8,'2021-02-02'),
+(5,6,'2020-01-05'),
+(5,6,'2020-3-8'),
+(5,6,'2020-03-14'),
+(3,7,'2021-05-4'),
+(9,8,'2021-2-24'),
+(7,6,'2019-12-21'),
+(7,5,'2020-8-10'),
+(7,6,'2021-4-7'),
+(10,7,'2019-9-29'),
+(8,8,'2020-10-3'),
+(8,8,'2020-11-4'),
+(4,6,'2019-1-24'),
+(4,6,'2019-5-15'),
+(4,6,'2020-2-27'),
+(4,6,'2020-8-3'),
+(6,7,'2020-5-24'),
+(6,5,'2021-1-11');
```

Comment:
- [ ] Because of the update on the `ids`, most data entry for the `animal_id` column of the table is wrong. For example, the current id of `Pikachu` is `3` while that of `Blossom` is `10`. In the data entry, the `id` of `Pikachu` is `5` while that of `Blossom` is `6`. 

The wrong data entry will make all your queries return the wrong results. I would suggest you automatically make data entries for the `animal_id` column of the `visits` table using the `SELECT` statement (``(SELECT id FROM animal WHERE name = 'Agumon')``) or you hardcode the correct ids. 👍 💻 

### Screenshot of the `animals` table in the database

![image](https://user-images.githubusercontent.com/77106962/230732625-ae49da5e-a26c-4f39-b9ed-59b92a6622ba.png)